src/acl/AnnotationData.cc

```diff
@@ -23,8 +24,9 @@ SBufList
 ACLAnnotationData::dump() const
 {
     SBufList sl;
-    if (const char *strNotes = notes->toString())
-        sl.push_back(SBuf(strNotes));
+    auto strNotes = notes->toString();
+    if (!strNotes.isEmpty())
+        sl.push_back(strNotes);
```

Comment:
Can we tell that notes are empty before printing them? Something along these lines, perhaps?

```suggestion
    if (!notes->empty())
        sl.push_back(notes->toString());
```

Reply:
Yep, you're right